x/group/types_test.go

```diff
@@ -0,0 +1,239 @@
+package group_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/x/group"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestPercentageDecisionPolicyAllow(t *testing.T) {
+	testCases := []struct {
+		name           string
+		policy         *group.PercentageDecisionPolicy
+		tally          *group.Tally
+		totalPower     string
+		votingDuration time.Duration
+		result         group.DecisionPolicyResult
+	}{
+		{
+			"YesCount percentage >= decision policy percentage",
+			&group.PercentageDecisionPolicy{
+				Percentage: "0.5",
+				Timeout:    time.Second * 100,
+			},
+			&group.Tally{
+				YesCount:     "2",
+				NoCount:      "0",
+				AbstainCount: "0",
+				VetoCount:    "0",
+			},
+			"3",
+			time.Duration(time.Second * 50),
+			group.DecisionPolicyResult{
+				Allow: true,
+				Final: true,
+			},
+		},
+		{
+			"YesCount percentage < decision policy percentage",
+			&group.PercentageDecisionPolicy{
+				Percentage: "0.5",
+				Timeout:    time.Second * 100,
+			},
+			&group.Tally{
+				YesCount:     "1",
+				NoCount:      "0",
+				AbstainCount: "0",
+				VetoCount:    "0",
+			},
+			"3",
+			time.Duration(time.Second * 50),
+			group.DecisionPolicyResult{
+				Allow: false,
+				Final: false,
+			},
+		},
+		{
+			"sum percentage < decision policy percentage",
+			&group.PercentageDecisionPolicy{
+				Percentage: "0.5",
+				Timeout:    time.Second * 100,
+			},
+			&group.Tally{
+				YesCount:     "1",
+				NoCount:      "2",
+				AbstainCount: "0",
+				VetoCount:    "0",
+			},
+			"3",
+			time.Duration(time.Second * 50),
+			group.DecisionPolicyResult{
+				Allow: false,
+				Final: true,
+			},
+		},
+		{
+			"sum percentage >= decision policy percentage",
+			&group.PercentageDecisionPolicy{
+				Percentage: "0.5",
+				Timeout:    time.Second * 100,
+			},
+			&group.Tally{
+				YesCount:     "0",
+				NoCount:      "0",
+				AbstainCount: "0",
+				VetoCount:    "0",
+			},
+			"3",
+			time.Duration(time.Second * 50),
+			group.DecisionPolicyResult{
+				Allow: false,
+				Final: false,
+			},
+		},
+		{
+			"decision policy timeout <= voting duration",
+			&group.PercentageDecisionPolicy{
+				Percentage: "0.5",
+				Timeout:    time.Second * 10,
+			},
+			&group.Tally{
+				YesCount:     "2",
+				NoCount:      "0",
+				AbstainCount: "0",
+				VetoCount:    "0",
+			},
+			"3",
+			time.Duration(time.Second * 50),
+			group.DecisionPolicyResult{
+				Allow: false,
+				Final: true,
+			},
```

Comment:
why is this `Allow: false`? Isn't the yes count > 50%?

Reply:
The decision policy timeout must be greater than the voting duration. In this case, it is less than the voting duration